Sources/Sentry/SentryCrashExceptionApplication.m

```diff
@@ -31,6 +33,12 @@ - (void)reportException:(NSException *)exception {
     }
     [super reportException:exception];
 }
+
+- (void)_crashOnException:(NSException *)exception {
```

Comment:
I could see a problem with this since there are 2 ways this method could be called. This is the stack trace of a regular `reportException` call

~~~
AppKit -[NSApplication _crashOnException:]
AppKit -[NSApplication reportException:]
...
~~~

So wouldn't this report the exception twice?

There are 2 possibilities that `_crashOnException` can be invoked

- Usually called by `reportException`
- Some AppKit APIs forgo that and directly call `_crashOnException`

So in this case we would have `self.reportException` -> `super.reportException` -> `self._crashOnException`

Reply:
@eaigner I sometimes receive crashes like (fetched from customers' machines, not uploaded to Sentry).

```
0   libsystem_kernel.dylib        	0x00007fff6b54033a __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff6b5fce60 pthread_kill + 430
2   libsystem_c.dylib             	0x00007fff6b4c7808 abort + 120
3   io.sentry.Sentry              	0x00000001034566be -[SentryCrashExceptionApplication _crashOnException:].cold.1 + 55
4   io.sentry.Sentry              	0x00000001034539d3 -[SentryCrashExceptionApplication _crashOnException:] + 12
5   com.apple.AppKit              	0x00007fff2eb10df6 -[NSApplication reportException:] + 901
6   io.sentry.Sentry              	0x000000010345399a -[SentryCrashExceptionApplication reportException:] + 371
7   com.apple.AppKit              	0x00007fff2ebc092d uncaughtErrorProc + 145
8   io.sentry.Sentry              	0x0000000103434679 handleException + 711
...
```

As you can see they are symbolicated but don't point to any of my code, only Sentry's. Do you think it's related to your observation above? If not, couldn't you point me to the right direction? Thanks :)